arkouda/numeric.py

```diff
@@ -474,21 +487,31 @@ def hash(
     cancel each other out, hence we do a rotation by the ordinal of
     the array.
     """
-    if isinstance(pda, pdarray):
-        return _hash_single(pda, full)
-
-    repMsg = type_cast(
-        str,
-        generic_msg(
-            cmd="efuncArr",
-            args={
-                "nameslist": [n.name for n in pda],
-                "typeslist": [n.objType for n in pda],
-                "length": len(pda),
-                "size": len(pda[0]),
-            },
-        ),
-    )
+    from arkouda import Categorical as _Categorical
+
+    if isinstance(pda, (pdarray, _Categorical, Strings)):
+        pda = _hash_helper(pda)
+        return pda.hash() if isinstance(pda, Strings) else _hash_single(pda, full)
+    elif isinstance(pda, List):
+        ready = [_hash_helper(a) for a in pda]
+        namesList = [n.name for n in ready]
+        typesList = [n.objType for n in ready]
+
+        repMsg = type_cast(
+            str,
+            generic_msg(
+                cmd="efuncArr",
```

Comment:
Why did we got with the command `efuncArr` here? I feel like `"hashlist"` or something like that would be more informative